src/doc/rustdoc/src/the-doc-attribute.md

```diff
@@ -214,3 +214,31 @@ the `strip-hidden` pass is removed.
 Since primitive types are defined in the compiler, there's no place to attach documentation
 attributes. This attribute is used by the standard library to provide a way to generate
 documentation for primitive types.
+
+## `#[cfg(rustdoc)]`: Documenting platform-/feature-specific information
```

Comment:
I just realized something about these docs. First, the heading here is a little misleading, since `cfg(rustdoc)` isn't just about platform-specific stuff. It's feasible that people could "invent" items that only exist for documentation, for whatever reason. However, writing something like "exposing items to documentation" feels *more* misleading, since it implies that you need that to add items to docs in the first place. I think this header text is acceptable, since platform-specific docs was the driving force for it, but it's something to consider.

Second, there isn't really a good chapter for these docs. This chapter is specifically about the `#[doc]` attribute, but this section is about something you add to the `#[cfg]` attribute. However, all the other chapters make even less sense to add this to. I'm wondering whether we should create a new chapter that talks about "advanced" features that are stable. I'm not sure what else would go with it, though... :thinking: Maybe we can just leave this here. I'm curious what @ollie27 and @steveklabnik think, though.